SDK Improvements

SergioCruzSL opened the pull request:
- GitIgnore adjustment;
- Messages now are available at the "OutputLog", just like the screen messages.
- Plugin option "CanContainContent" changed to true, then we will be able to send the template blueprint inside the plugin.
- "UCharisma" split into "UCharismaAPI" and "UPlaythrough", as required at the "Unreal SDK Improvement" document.
- Creation of the Blueprint Async for the API initialization, and further cleanup the CPP classes.
- Implemented the "Save Emotions/Memories" system, as required at the SDK Improvements list.
- Previous Pull request comments corrections.
- Creation of the templates and Charisma components.

SergioCruzSL committed:
UPDATE

Git ignore adjustment.

SergioCruzSL committed:
UPDATE

Git ignore adjustment.

SergioCruzSL committed:
UPDATE

Messages now are available at the "OutputLog", just like the screen messages.

SergioCruzSL committed:
UPDATE

Plugin option "CanContainContent" changed to true, then we will be able to send the template blueprint inside the plugin.

SergioCruzSL committed:
UPDATE

UCharisma divided into UCharismaAPI and UPlaythrough, as required at the "Unreal  SDK Improvement" document.

SergioCruzSL committed:
UPDATE

Creation of the Blueprint Async for the API initialization, and further cleanup the CPP classes.

SergioCruzSL committed:
UPDATE

Creation of the Blueprint Async Nodes to broadcast playthrough events.

SergioCruzSL committed:
UPDATE

Adjustment at the UPlaythrough and UCharismaAPI, moving functions from Playthrough to API, changing them to static (As requested at Discord).

bensalilijames commented on Source/CharismaModule/Private/CharismaAPI.cpp:
```diff
@@ -0,0 +1,196 @@
+	return Result;
+}
+
+void UCharismaAPI::CreateCharismaPlaythroughObject(UPlaythrough*& Playthrough)
+{
+	Playthrough = NewObject<UPlaythrough>();
+}
```
This could make more sense to be on the `UPlaythrough` class as `UPlaythrough::CreatePlaythroughObject`, as it's more to do with the Playthrough than the API 🙂

bensalilijames commented on Source/CharismaModule/Private/CharismaInitialization.cpp:
```diff
@@ -0,0 +1,44 @@
+
+	UCharismaAPI::CreateCharismaPlaythroughObject(RefPlaythrough);
+
+	RefPlaythrough->OnTokenCreated.AddDynamic(this, &UCharismaInitialization::BroadcastSuccess);
+	RefPlaythrough->OnTokenFailCreation.AddDynamic(this, &UCharismaInitialization::BroadcastFailure);
```
Really minor, but to be consistent with Blueprint `BroadcastSuccess` and `BroadcastFailure`, perhaps we can use `OnTokenCreationSuccess` and `OnTokenCreationFailure`?

bensalilijames commented on Source/CharismaModule/Private/PlaythroughStartConversation.cpp:
```diff
@@ -0,0 +1,39 @@
+// Fill out your copyright notice in the Description page of Project Settings.
```
I totally get where this is coming from, though I think attaching handlers to the `Playthrough` is a distinct step from starting the playthrough so I'm not sure they should be related.

Perhaps instead of a `PlaythroughStartConversation` async node, it could be an `AttachPlaythroughHandlers` async node, which has the `Playthrough` as an input pin, and lots of async output pins, one for each event such as `OnReady`, `OnMessage`, `OnTyping` etc? What do you think?

SergioCruzSL committed:
UPDATE

Implemented the "Save Emotions/Memories" system, as required at the SDK Improvements list.

SergioCruzSL committed:
UPDATE

Charisma components, and templates created.

SergioCruzSL committed:
UPDATE

Last commit improvement, and adjustments of "CreateCharismaPlaythroughObject", "OnTokenCreationSuccess" and "OnTokenCreationFailure", just like required at the pull request observations.

SergioCruzSL committed:
UPDATE

Creation of the "AttachPlaythroughHandlers" async node, as required at the pull request comments.

SergioCruzSL committed:
UPDATE

Implementation of the audio option inside "CharismaActorComponent".

SergioCruzSL committed:
UPDATE

Changed "BP_CharismaTemplateGameMode" to "BP_CharismaGameModeTemplate".

SergioCruzSL committed:
UPDATE

Audio system inside the "CharismaActorComponent" updated, as required by Discord.

SergioCruzSL committed:
UPDATE

Reconnection system update.
